examples/cameoparison-svelte-machine/src/machine.js

```diff
@@ -0,0 +1,204 @@
+import { assign, createMachine, send } from 'xstate';
+
+import { ROUNDS_PER_GAME } from './constants.js';
+import { select } from './select.js';
+import { loadImage } from './utils.js';
+
+const loadCelebs = async () => {
+  const res = await fetch('https://cameo-explorer.netlify.app/celebs.json');
+  const data = await res.json();
+
+  const lookup = new Map();
+  data.forEach((c) => {
+    lookup.set(c.id, c);
+  });
+
+  const subset = new Set();
+  data.forEach((celeb) => {
+    if (celeb.reviews >= 50) {
+      subset.add(celeb);
+      celeb.similar.forEach((id) => {
+        subset.add(lookup.get(id));
+      });
+    }
+  });
+
+  return {
+    celebs: Array.from(subset),
+    lookup
+  };
+};
+
+const loadRounds = (selection) => {
+  return Promise.all(selection.map((round) => loadCelebPair(round)));
+};
```

Comment:
This should be (and all the consequences need to be implemented):
```js
const loadRounds = (selection) => selection.map((round) => loadCelebPair(round));
```

The example works as it is, but I will improve it in the next weeks. My first approach was to put all the logic into the machines, and now I need to check what should be moved back into the components.

Same goes with the other example with actors.